diff --git a/javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/AbstractSymbolResolutionTest.java b/javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/AbstractSymbolResolutionTest.java
index 52bc79d7b..37907b6c3 100644
--- a/javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/AbstractSymbolResolutionTest.java
+++ b/javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/AbstractSymbolResolutionTest.java
@@ -66,7 +66,13 @@ public abstract class AbstractSymbolResolutionTest {
         JDK15(15),
         JDK16(16),
         JDK17(17),
-        JDK18(18);
+        JDK18(18),
+        JDK19(19),
+        JDK20(20),
+        JDK21(21),
+        JDK22(22),
+        JDK23(23),
+        JDK24(24);
 
         private final Integer major;
 
@@ -116,9 +122,22 @@ public abstract class AbstractSymbolResolutionTest {
                 return JDK17;
             } else if ("18".equals(javaVersion) || javaVersion.startsWith("18.")) {
                 return JDK18;
+            } else if ("19".equals(javaVersion) || javaVersion.startsWith("19.")) {
+                return JDK19;
+            } else if ("20".equals(javaVersion) || javaVersion.startsWith("20.")) {
+                return JDK20;
+            } else if ("21".equals(javaVersion) || javaVersion.startsWith("21.")) {
+                return JDK21;
+            } else if ("22".equals(javaVersion) || javaVersion.startsWith("22.")) {
+                return JDK22;
+            } else if ("23".equals(javaVersion) || javaVersion.startsWith("23.")) {
+                return JDK23;
+            } else if ("24".equals(javaVersion) || javaVersion.startsWith("24.")) {
+                return JDK24;
             }
 
-            throw new IllegalStateException("Unable to determine the current version of java running");
+            throw new IllegalStateException(
+                    "Unable to determine the current version of java running from: " + javaVersion);
         }
 
         /**
